src/operator/tensor/matrix_op-inl.h

```diff
@@ -257,6 +257,29 @@ struct TransposeParam : public dmlc::Parameter<TransposeParam> {
   }
 };
 
+
+template<typename xpu, typename DType>
+MSHADOW_XINLINE void Transpose2D(DType *in, DType *out, index_t shape_0, index_t shape_1) {
+// ensure cache line hits and prevent cache miss for any configuration
+index_t blocksize = 32;
+index_t n = shape_0;
+index_t p = shape_1;
+
+for (index_t i = 0; i < n; i += blocksize) {
+  #pragma omp parallel for
+    for (index_t j = 0; j < p; j += blocksize) {
+        // transpose the block
+        #pragma unroll 4
```

Comment:
add comment explaining why unroll by 4 ?

Reply:
Do we really have to unroll this mannually? I thought the compiler could figure out the optimal unrolling factor.

Reply:
Since there is not much perf difference for limited tests ran, removing the explicit/manual unroll.